### ISSUE TITLE
Suggestion: provide built-in rulesets with good defaults

### ISSUE THREAD

**Issue by alefranz** [DotNetAnalyzers/StyleCopAnalyzers#3194]
Adding this analyzers to a project is overwhelming due to the amount of warnings usually generated and tweaking the rules is a time consuming job that requires reading through the docs and multiple GitHub issues to understand implications.

It would be nice to have some built-in rulesets with sane defaults so that analysers can easily be enable in projects without having to spend countless hours tweaking when you are ok with common current practices, as well as not having to add both a ruleset and a stylecop.json.

I am not suggesting to make this a default behaviour as that would be a breaking change, but having an easy and well documented way to enable this ruleset.

I imagine there should be at least 2 rulesets: 
- one for applications and internal libraries
- one for distributed libraries (which also enables SA0001 and SA16xx)

This is for example the minimal ruleset currently required to remove warnings for the ASP.NET Core API template:

ruleset
```xml
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="Default" ToolsVersion="16.0">
  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
    <Rule Id="SA0001" Action="None" />
    <Rule Id="SA1101" Action="None" />
    <Rule Id="SA1309" Action="None" />
    <!-- https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3193 -->
    <Rule Id="SA1413" Action="Info" />
    <Rule Id="SA1600" Action="None" />
    <Rule Id="SA1601" Action="None" />
    <Rule Id="SA1602" Action="None" />
    <Rule Id="SA1604" Action="None" />
    <Rule Id="SA1605" Action="None" />
    <Rule Id="SA1606" Action="None" />
    <Rule Id="SA1607" Action="None" />
    <Rule Id="SA1608" Action="None" />
    <Rule Id="SA1610" Action="None" />
    <Rule Id="SA1611" Action="None" />
    <Rule Id="SA1612" Action="None" />
    <Rule Id="SA1613" Action="None" />
    <Rule Id="SA1614" Action="None" />
    <Rule Id="SA1615" Action="None" />
    <Rule Id="SA1616" Action="None" />
    <Rule Id="SA1617" Action="None" />
    <Rule Id="SA1618" Action="None" />
    <Rule Id="SA1619" Action="None" />
    <Rule Id="SA1620" Action="None" />
    <Rule Id="SA1621" Action="None" />
    <Rule Id="SA1622" Action="None" />
    <Rule Id="SA1623" Action="None" />
    <Rule Id="SA1624" Action="None" />
    <Rule Id="SA1625" Action="None" />
    <Rule Id="SA1626" Action="None" />
    <Rule Id="SA1627" Action="None" />
    <Rule Id="SA1629" Action="None" />
    <Rule Id="SA1633" Action="None" />
    <Rule Id="SA1634" Action="None" />
    <Rule Id="SA1635" Action="None" />
    <Rule Id="SA1636" Action="None" />
    <Rule Id="SA1637" Action="None" />
    <Rule Id="SA1638" Action="None" />
    <Rule Id="SA1640" Action="None" />
    <Rule Id="SA1641" Action="None" />
    <Rule Id="SA1642" Action="None" />
    <Rule Id="SA1643" Action="None" />
    <Rule Id="SA1648" Action="None" />
    <Rule Id="SA1649" Action="None" />
    <Rule Id="SA1651" Action="None" />
    <Rule Id="SX1101" Action="Warning" />
    <Rule Id="SX1309" Action="Warning" />
    <Rule Id="SX1309S" Action="Warning" />
  </Rules>
</RuleSet>
```

and the the required stylecop.json
```json
{
  "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
  "settings": {
    "orderingRules": {
      "usingDirectivesPlacement": "outsideNamespace"
    }
  }
}
```

I don't think SA1413 should be in it, but the template is affected by #3193

**Comment by sharwell** [comment]
The default value for each rule is already configured according to the recommended defaults. Users typically disable rules if/when they see warnings during installation, and this is always a team-specific evaluation of the rules against the policies used by that team.

**Comment by sharwell** [comment]
We already have a default ruleset for reference here:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/rulesets/StyleCopAnalyzersDefault.ruleset

I'll file a follow-up issue to add an equivalent global .editorconfig.

**Comment by sharwell** [comment]
Closed as **wontfix** since we don't currently have plans to create "profiles" of rules, but opened a related issue to simplify the customization process with newer compilers.
